scraper/src/fcc2zim/scraper.py

```diff
@@ -0,0 +1,164 @@
+        self.long_description = long_description
+        (self.description, self.long_description) = compute_descriptions(
+            self.description, self.description, self.long_description
+        )
+
+        self.content_creator = content_creator
+        self.publisher = publisher
+        self.force = force
+        self.course_csv = course_csv
+        if not zip_path:
+            self.zip_path = self.build_dir.joinpath("main.zip")
+        else:
+            self.zip_path = Path(zip_path)
+            if not self.zip_path.exists():
+                raise ValueError(f"Zip file not found in {self.zip_path}")
+
+        # if we do not build the ZIM, we can stop here
+        if not self.do_build:
+            return
+
+        period = start_date.strftime("%Y-%m")
+        if zim_file:
+            self.zim_path = Path(zim_file.format(period=period))
+            # make sure we were given a filename and not a path
+            if Path(self.zim_path.name) != self.zim_path:
+                raise ValueError(f"zim_name is not a filename: {zim_file}")
+        else:
+            self.zim_path = Path(f"{name}_{period}.zim")
+
+        # build full path
+        self.zim_path = self.output_dir.joinpath(self.zim_path)
+
+        if self.zim_path.exists():
+            if not self.force:
+                raise ValueError(f"ZIM file {self.zim_path} already exist.")
+            Global.logger.info(f"Removing existing ZIM file {self.zim_path}")
+            self.zim_path.unlink()
+        else:
+            Global.logger.info(f"ZIM path: {self.zim_path}")
+
+        logo_path = Path(__file__).parent.joinpath("assets", "fcc_48.png")
+        if not logo_path.exists():
+            raise ValueError(f"Logo not found at {logo_path}")
+
+        self.creator = Creator(self.zim_path, "index.html").config_metadata(
+            Name=self.name,
+            Title=self.title,
+            Publisher=self.publisher,
+            Date=start_date,
+            Creator=self.content_creator,
+            Description=self.description,
+            LongDescription=self.long_description,
+            Language=self.language,
+            Tags=";".join(["FCC", "freeCodeCamp"]),
+            Scraper=f"fcc2zim v{VERSION}",
+            Illustration_48x48_at_1=logo_path.read_bytes(),
+        )
+
+        # start creator early to detect any problem early as well
+        self.creator.start()
+
+    def run(self):
+        try:
+            self.run_commands()
+        except Exception as exc:
+            if self.creator:
+                self.creator.can_finish = False
+            if isinstance(exc, KeyboardInterrupt):
+                Global.logger.error("KeyboardInterrupt, exiting.")
+                raise SystemExit(3) from exc
+            else:
+                Global.logger.error(f"Interrupting process due to error: {exc}")
+                Global.logger.exception(exc)
+                raise SystemExit(2) from exc
+        else:
+            if self.creator:
```

Comment:
If not `do_build` then this doesn't exists…

Reply:
Yes, and this is why I check that it exists. And I don't care if it is due to `do_build` flag value or anything else which prevented the existence of this value. What do you suggest, using `if do_build:`?

Reply:
No, `self.creator` simply doesn't exists so `if self.creator` will raise `NameError`. You can just set it to `None` early on